### DISCUSSION TITLE
[ 功能建议 ] Binary 数据类型支持显示为原本的HEX字符串

### DISCUSSION THREAD

**Discussion by beebird** [hhyo/Archery#1766] [Ideas]
* Binary data 显示的不具有可读性:
```sql
 select id, uid, account from `table1` order by id limit 1;
```
    2 | 000001  | yû÷¼þ.F§	Â¼§Ýæ

* 目前只能通过 `HEX` 函数:
```sql
 select id, uid, HEX(account) from `table1` order by id limit 1;
```
    2 | 000001 | 9D79FAF7BCFE122E46A81809C2BCA7A3FD8BE681

如果可以默认将 Binary 格式的HEX 数据，以可读的形式显示出来，就是一个很受欢迎的feature了。哪怕只支持少数数据库。

  参考:  

-  [[MySQL] Improved Binary data type support that shows the actual HEX value and you can edit directly.](https://tableplus.com/blog/2019/10/whats-new-in-build-270.html) 

-  https://github.com/TablePlus/TablePlus/issues/1439

**Comment by LeoQuote** [comment]
我们也有用户反馈说有需要这个, 这个改起来不会太难, 只要改一下 json decoder 就可以了, 我帮你转成 issue , 欢迎 pr

**Reply by beebird** [reply]
似乎比预想的麻烦，除了Binary字段类型，其他某些字段也会返回为unicode字符串。似乎只能在查询时候处理，拿到result再处理就容易把其他不该转换的字段也转换了。mysql客户端有个参数是 [--binary-as-hex](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex)，试了下好像[MySQLdb](https://mysqlclient.readthedocs.io/index.html) 不支持 @LeoQuote

**Reply by LeoQuote** [reply]
那就没有办法做了, 即使 mysqldb 支持, 那也需要知道查询出来的字段类型, 目前 archery 没有这种机制去判断这个
